nifi-commons/nifi-write-ahead-log/src/main/java/org/wali/MinimalLockingWriteAheadLog.java

```diff
@@ -474,8 +474,15 @@ private void recoverFromEdits(final Map<Object, T> modifiableRecordMap, final Lo
                 subsequentTransactionId = nextPartition.getNextRecoverableTransactionId();
             } catch (final IOException e) {
                 logger.error("{} unexpectedly found End-of-File when reading from {} for Transaction ID {}; "
-                        + "assuming crash and ignoring this transaction",
+                        + "attempting to get the next recoverable transaction ID",
```

Comment:
@gresockj thanks for the contribution! The only concern that I have here is that this update assumes that we will never get more than 1 IOException in a row... I think we need to continue looping until we've run out of 'recovery files' for the partition.

Reply:
I had a thought that this would be necessary, but I'm not sure what the ending condition would be.  How would you determine that we've run out of recovery files?